components/templates/McGonagall/McGonagall.tsx

```diff
@@ -101,21 +243,30 @@ class McGonagall extends React.Component {
    * @param {object} currXState current xstate object
    * @returns {object} updated currXstate, activeCard, and cardHistory
    */
-  navigateToLatestCard(cardHistory, currXState) {
+  navigateToLatestCard(
+    cardHistory: Step[],
+    currXState: State<TContext, McGonagallEvent, any, any>,
+  ): {
+    currXState: State<TContext, McGonagallEvent, any, any>;
+    activeCard: string;
+    cardHistory: Step[];
+  } {
     const newCardHistory = cardHistory;
+
     let latest = this.stateMachine.transition(currXState.value, {
       type: 'NEXT',
       navigatingToLatestCard: true,
       ...currXState.context,
     });
+
     let finalReached = false;
 
     do {
       if (
         latest.value !== currXState.value &&
         latest.value !== newCardHistory[0].name
       ) {
-        newCardHistory.unshift(this.getMatchingStep(this.props.steps, latest));
+        newCardHistory.unshift(this.getMatchingStep(this.props.steps, latest)!);
```

Comment:
Based on the signature of `getMatchingStep` I don't think this `!` is necessary.

```suggestion
        newCardHistory.unshift(this.getMatchingStep(this.props.steps, latest));
```